includes/Utilities/Plugin_Request_Utility.php

```diff
@@ -181,4 +181,79 @@ public static function get_files_to_ignore() {
 
 		return $files_to_ignore;
 	}
+
+	/**
+	 * Downloads the plugin from the URL and installs it.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param string $plugin_url The URL of the plugin to download.
+	 * @return string|bool The plugin basename if the plugin is downloaded and installed successfully, false otherwise.
+	 */
+	public static function download_plugin( $plugin_url ) {
```

Comment:
Could this use the `download_url()` function provided by WordPress? And then maybe `wp_handle_sideload()` with the custom upload dir. This would allow removing a lot of code here and we wouldn't have to worry about `WP_Filesystem` etc.

Reply:
I didn't know that function. For sure, I'll check it and propose a change.

Reply:
It's giving me a fatal error download_url, even if I use the require...

Reply:
PHP Fatal error:  Uncaught Error: Call to undefined function WordPress\Plugin_Check\Utilities\download_url() in /wp-content/plugins/plugin-check/includes/Utilities/Plugin_Request_Utility.php:194

Reply:
@davidperezgar I am checking it.